src/Dialog.js

```diff
@@ -45,11 +46,14 @@ const UnstyledButton = styled(Flex).attrs({
   background: none;
   border: none;
   padding: 0;
+
+  position: absolute;
+  top: 16px;
+  right: 16px;
 `
 
-const DialogHeader = styled(Flex).attrs({
+const DialogHeaderBase = styled(Flex).attrs({
```

Comment:
Looks like setting these values via `attrs` means that user provided values for these don't get applied. Since this is now an exposed component we need to move these three values (p, justifyContent, and alignItems) to default props down on line 110.

This is probably the best approach going forward for the ol' styles getting overridden by extended components issue - if you set all the values that belong in system props by adding them to defaultProps that insures that they will be overridden properly if a user provides a value. The downside is that if you can't pull from the theme to create a value in defaultProps 

for example if you wouldn't be able to do:

DialogHeader.defaultProps = {
 p: theme.spacing[3]
}

because theme here would always be _our_ default theme and wouldn't respect a custom theme provided 🤔